util/len-caching-lock/src/mutex.rs

```diff
@@ -13,30 +13,15 @@
 
 // You should have received a copy of the GNU General Public License
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
-extern crate parking_lot;
 
-use std::collections::VecDeque;
 use std::ops::{Deref, DerefMut};
 use std::sync::atomic::AtomicUsize;
 use std::sync::atomic::Ordering;
 
 use parking_lot::{Mutex, MutexGuard};
 
-/// Implement to allow a type with a len() method to be used
-/// with [`LenCachingMutex`](struct.LenCachingMutex.html)
-pub trait Len {
-	fn len(&self) -> usize;
-}
-
-impl<T> Len for Vec<T> {
-	fn len(&self) -> usize { self.len() }
-}
-
-impl<T> Len for VecDeque<T> {
-	fn len(&self) -> usize { self.len() }
-}
-
-/// Can be used in place of a `Mutex` where reading `T`'s `len()` without 
+use Len;
+/// Can be used in place of a [`Mutex`](../../lock_api/struct.Mutex.html) where reading `T`'s `len()` without 
 /// needing to lock, is advantageous. 
 /// When the Guard is released, `T`'s `len()` will be cached.
 /// The cached `len()` may be at most 1 lock behind current state.
```

Comment:
Should have a `Default` and non-blocking `Debug` implementation.

Reply:
Also added `From` implementation